crates/typst/src/model/content.rs

```diff
@@ -2,168 +2,165 @@ use std::any::TypeId;
-/// content is expected, you can also pass a [string]($str) or `{none}`.
-///
-/// # Representation
-/// Content consists of elements with fields. When constructing an element with
-/// its _element function,_ you provide these fields as arguments and when you
-/// have a content value, you can access its fields with [field access
-/// syntax]($scripting/#field-access).
-///
-/// Some fields are required: These must be provided when constructing an
-/// element and as a consequence, they are always available through field access
-/// on content of that type. Required fields are marked as such in the
-/// documentation.
-///
-/// Most fields are optional: Like required fields, they can be passed to the
-/// element function to configure them for a single element. However, these can
-/// also be configured with [set rules]($styling/#set-rules) to apply them to
-/// all elements within a scope. Optional fields are only available with field
-/// access syntax when they are were explicitly passed to the element function,
-/// not when they result from a set rule.
-///
-/// Each element has a default appearance. However, you can also completely
-/// customize its appearance with a [show rule]($styling/#show-rules). The show
-/// rule is passed the element. It can access the element's field and produce
-/// arbitrary content from it.
-///
-/// In the web app, you can hover over a content variable to see exactly which
-/// elements the content is composed of and what fields they have.
-/// Alternatively, you can inspect the output of the [`repr`]($repr) function.
 #[ty(scope)]
-#[derive(Debug, Clone, Hash)]
-#[allow(clippy::derived_hash_with_manual_eq)]
-pub struct Content {
-    elem: Element,
-    attrs: EcoVec<Attr>,
+#[repr(transparent)]
+#[derive(Debug, Clone)]
+pub struct Content(pub Arc<dyn Element>);
+
+impl Default for Content {
+    fn default() -> Self {
+        Self::empty()
+    }
 }
 
-/// Attributes that can be attached to content.
-#[derive(Debug, Clone, PartialEq, Hash)]
-enum Attr {
-    Span(Span),
-    Field(EcoString),
-    Value(Prehashed<Value>),
-    Child(Prehashed<Content>),
-    Styles(Styles),
-    Prepared,
-    Guard(Guard),
-    Location(Location),
+impl<T: Element> From<T> for Content {
+    fn from(value: T) -> Self {
+        Self::new(value)
+    }
+}
+
+impl From<Arc<dyn Element>> for Content {
+    fn from(value: Arc<dyn Element>) -> Self {
+        Self(value)
+    }
 }
 
 impl Content {
-    /// Create an empty element.
-    pub fn new(elem: Element) -> Self {
-        Self { elem, attrs: EcoVec::new() }
+    #[inline]
+    pub fn new<E: Element>(elem: E) -> Self {
+        Self(Arc::new(elem))
     }
 
-    /// Create empty content.
+    #[inline]
     pub fn empty() -> Self {
-        Self::new(SequenceElem::elem())
+        Self::new(SequenceElem::default())
+    }
+
+    pub fn temp(of: ElementData) -> Self {
+        of.empty()
+    }
```

Comment:
From a quick search, it seems like the whole `#[empty]` business is only necessary to support this function, which is only used once to get the local name of a figure without making one. There's gotta be a better way. I think empty elements just shouldn't be a thing.

Reply:
Ok, I've now seen that you actually said this in the initial PR description. Then, I think we should _definitely_ find another way to handle the figure thing.

Reply:
Yep we do, but how? 😭